app/assets/javascripts/lessons.js

```diff
@@ -1,121 +1,120 @@
-      checkbox.removeClass('lc-unchecked');
-      checkbox.removeClass('lc-checked');
-      checkbox.addClass('lc-waiting');
-    
-      // submit the form
-      checkbox.parents('form').submit();
-  });
+var LESSON_HEADINGS = [
+  'Introduction',
+  'Learning Outcomes',
+  'Assignment',
+  'Additional Resources'
+];
 
-  // Delegate a hover listener (separate mouseenter and mouseleave)
-  // to completion wrapper and don't enable if it's a disabled class
-  $('.lesson').on('mouseenter', '.lc-checkbox',function(e){
-    checkbox = $(e.target).parents('.lesson').find('.lc-checkbox');
-    
-    if(!checkbox.hasClass('lc-disabled')){
-      checkbox.addClass('lc-hover');
-    }
-  });
-  $('.lesson').on('mouseleave', '.lc-checkbox',function(e){
-    checkbox = $(e.target).parents('.lesson').find('.lc-checkbox');
-    
-    if(!checkbox.hasClass('lc-disabled')){
-      checkbox.removeClass('lc-hover');
-    }
-  });
+function getElements(selector) {
+  return document.querySelectorAll(selector);
+}
 
-  // INDIVIDUAL LESSON SHOW PAGES
-
-  // Delegate listeners to the wrapper so they stick around when new forms
-  // and checkboxes are created.  This is an example of "event delegation".
-  $('.lc-end-wrapper').on('click', '.checkbox-container', function(e){
-    if( !$('.checkbox-container').hasClass('lc-disabled') &&
-        !$('.checkbox-container').hasClass('lc-checked') ){
-      e.preventDefault();
-      checkboxContainer = $('.checkbox-container');
-      
-      // Switch to AJAX waiting state regardless of prior state
-      // since it will be overwritten anyway
-      // by the refresh from the unobtrusive javascript
-      checkboxContainer.addClass('lc-waiting');
-      
-      // submit the form
-      $(e.target).parents('form').submit();
-    }
+function kebabCase(text) {
+  return text.toLowerCase().match(/\w+/g).join('-');
+}
+
+function setTargetForExternalLinks() {
+  getElements('.lesson-content a[href^=http]').forEach(function(externalLink) {
+    externalLink.setAttribute('target', '_blank');
   });
-    
-  // Delegate a hover listener (separate mouseenter and mouseleave) to completion wrapper
-  // Don't enable if it's a disabled class
-  $('.lc-end-wrapper').on('mouseenter', '.checkbox-container',function(e){
-    if(!$('.checkbox-container').hasClass('lc-disabled')){
-      $('.checkbox-container').addClass('lc-hover');
+}
+
+function navigationElement(headingText) {
+  return (
+    '<div class="lesson-navigation__item">' +
+    '<div class="lesson-navigation__circle"></div>' +
+    '<div class="lesson-navigation__title">' +
+    '<a class="lesson-navigation__link grey" href="#' + kebabCase(headingText) + '" data-turbolinks="false">' + headingText +
+    '</a></div></div>'
+  );
+}
+
+function lessonNavigation(headings) {
+  return headings.map(navigationElement).join('');
+}
+
+function getInnerText(heading) {
+  return heading.innerText;
+}
+
+function isCommonHeading(heading) {
```

Comment:
Couldn't come up with a better function name, can you suggest one? @KevinMulhern

Reply:
This looks fine to me 👍